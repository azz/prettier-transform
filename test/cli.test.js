"use strict";

describe("cli", () => {
  test("applies a transform to a file", () => {
    const result = run([
      "-t=./test/transform-foo-to-bar",
      "--parser=babylon",
      "./test/foo.js"
    ]);
    expectPass(result);
    expect(result.output.toString()).toMatchSnapshot();
  });

  test("passes unknown options to prettier", () => {
    const result = run([
      "-t=./test/transform-foo-to-bar",
      "--parser=babylon",
      "./test/foo.js",
      "--no-semi"
    ]);
    expectPass(result);
    expect(result.output.toString()).toMatchSnapshot();
  });
});

const spawnSync = require("child_process").spawnSync;

function run(args) {
  return spawnSync("node", ["./bin/prettier-transform.js"].concat(args));
}

function expectPass(result) {
  expect(result.error).toBeFalsy();
  expect(result.stderr.toString()).toEqual("");
}
