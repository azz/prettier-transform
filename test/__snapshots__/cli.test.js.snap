// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli applies a transform to a file 1`] = `
",/* eslint-disable */
bar();

,"
`;

exports[`cli passes unknown options to prettier 1`] = `
",/* eslint-disable */
bar()

,"
`;
